src/huggingface_hub/hf_api.py

```diff
@@ -1094,82 +1127,80 @@ def model_info(
         return ModelInfo(**d)
 
     @_deprecate_positional_args
-    def list_repo_files(
+    def dataset_info(
         self,
         repo_id: str,
         *,
         revision: Optional[str] = None,
-        repo_type: Optional[str] = None,
         token: Optional[str] = None,
         timeout: Optional[float] = None,
-    ) -> List[str]:
+    ) -> DatasetInfo:
         """
-        Get the list of files in a given repo.
+        Get info on one specific dataset on huggingface.co
+
+        Dataset can be private if you pass an acceptable token.
 
         Args:
             repo_id (`str`):
                 A namespace (user or an organization) and a repo name separated
                 by a `/`.
             revision (`str`, *optional*):
-                The revision of the model repository from which to get the
+                The revision of the dataset repository from which to get the
                 information.
-            repo_type (`str`, *optional*):
-                Set to `"dataset"` or `"space"` if uploading to a dataset or
-                space, `None` or `"model"` if uploading to a model. Default is
-                `None`.
             token (`str`, *optional*):
                 An authentication token [1]_.
```

Comment:
This reference is not rendered properly ([link](https://moon-ci-docs.huggingface.co/docs/huggingface_hub/pr_792/en/package_reference/hf_api#huggingface_hub.HfApi.dataset_info) ). How do we do references in mdx?

Reply:
I don't think there's a mechanism in place yet, I've replaced occurrences with a link directly.

Reply:
opened an issue for this: https://github.com/huggingface/doc-builder/issues/181